etsin_finder/frontend/js/components/general/navigation/langToggle.jsx

```diff
@@ -10,83 +10,59 @@
 import styled from 'styled-components'
 import translate from 'counterpart'
 
 import { TransparentButton, InvertedButton } from '../button'
 
-class LangToggle extends Component {
-  static propTypes = {
-    inverted: PropTypes.bool,
-    margin: PropTypes.string,
-    Stores: PropTypes.object.isRequired,
+const LangToggle = ({ inverted, margin, Stores }) => {
+  const { Locale } = Stores
+  const changeLang = () => {
+    Locale.toggleLang()
   }
 
-  static defaultProps = {
-    inverted: false,
-    margin: '0.3em 0.3em',
-  }
-
-  state = {
-    announce: '',
-  }
-
-  changeLang = () => {
-    this.props.Stores.Locale.toggleLang()
-    setTimeout(() => {
-      this.setState({
-        announce: translate('general.state.changedLang', {
-          lang: this.props.Stores.Locale.currentLang,
-        }),
-      })
-    }, 50)
-    setTimeout(() => {
-      this.setState({
-        announce: '',
-      })
-    }, 500)
-  }
-
-  otherLang = () => this.props.Stores.Locale.languages.map(lang => {
-      if (lang !== this.props.Stores.Locale.currentLang) {
+  const otherLang = () =>
+    Locale.languages.map(lang => {
+      if (lang !== Locale.currentLang) {
         return <Lang key={lang}>{lang}</Lang>
       }
       return null
     })
 
-  render() {
-    return (
-      <div>
-        <div className="sr-only" aria-live="assertive">
-          {this.state.announce}
-        </div>
-        <div>
-          <span className="sr-only">
-            {translate('general.language.toggleLabel')}
-            :
-            {this.otherLang()}
-          </span>
-          {this.props.inverted ? (
-            <InvertedButton
-              color="dark"
-              margin={this.props.margin}
-              padding="0.3em 1em 0.4em"
-              onClick={this.changeLang}
-            >
-              {this.otherLang()}
-            </InvertedButton>
-          ) : (
-            <TransparentButton onClick={this.changeLang} margin={this.props.margin}>
-              {this.otherLang()}
-            </TransparentButton>
-          )}
-        </div>
-      </div>
-    )
-  }
+  return (
+    <>
+      <span className="sr-only" id="sr-text-for-language-toggle">
+        {translate('general.language.toggleLabel', {
+          otherLang:
+            Locale.currentLang === 'fi'
+              ? translate('qvain.general.langEn')
+              : translate('qvain.general.langFi'),
+        })}
+      </span>
+      {inverted ? (
+        <InvertedButton
+          color="dark"
+          margin={margin}
+          padding="0.3em 1em 0.4em"
+          onClick={changeLang}
+          aria-labelledby="sr-text-for-language-toggle"
+        >
+          {otherLang()}
+        </InvertedButton>
+      ) : (
+        <TransparentButton
+          onClick={changeLang}
+          margin={margin}
+          aria-labelledby="sr-text-for-language-toggle"
+        >
+          {otherLang()}
+        </TransparentButton>
+      )}
+    </>
+  )
```

Comment:
At least in ChomeVox this causes "Switch language to: Finnish" to be spoken twice for me. First, the sr-only `div` and then the button label. Instead of using the sr-only `div`, translating the `aria-label` attribute directly should avoid this.

```suggestion
  const otherLangName =
    Locale.currentLang === 'fi'
      ? translate('qvain.general.langEn')
      : translate('qvain.general.langFi')

  return (
    <>
      {inverted ? (
        <Translate
          component={InvertedButton}
          color="dark"
          margin={margin}
          padding="0.3em 1em 0.4em"
          onClick={changeLang}
          attributes={{ 'aria-label': 'general.language.toggleLabel' }}
          with={{ otherLang: otherLangName }}
        >
          {otherLang()}
        </Translate>
      ) : (
        <Translate
          component={TransparentButton}
          onClick={changeLang}
          margin={margin}
          attributes={{ 'aria-label': 'general.language.toggleLabel' }}
          with={{ otherLang: otherLangName }}
        >
          {otherLang()}
        </Translate>
      )}
```
  
```

Reply:
This suggestion causes the same problem mentioned earlier with aria-label, "Used 'aria-labelledby'
instead of 'aria-label' because the sr would not read 'aria-label' in
correct language".

Reply:
After testing with Firefox it seems like no text is read in the current language, only in English. Also, the aria-live text is not announced when language is switched... Must see what can be done about that

Reply:
Seems like it is a bug in Firefox that has been there for over 5 years... I guess we could make the conclusion to support other browsers over firefox at least when it comes to Voiceover.